### DISCUSSION TITLE
Naming and responsabilities

### DISCUSSION THREAD

**Discussion by bilboquet** [massalabs/massa-proto#2] [General]
@aoudiamoncef @ASAPSegfault 

We agreed that this repo should be named `mass-proto` and aim to centralize and factorize ou `.proto` files.

We agreed that the bindings for different languages we support should be in separate repositories.

I would like to suggest that we use the following convention for the repositories that contain the bindings (that's very close to what we discussed):
`massa-proto-bindings-rs` for Rust bindings
`massa-proto-bindings-go` for Go bindings
`massa-proto-bindings-as`for AssemblyScript bindings

I just added `bindings`, but I found it better this way.

I have a question:
At least on the Rust side (and probably on the AS side too) we will soon have bindings both for the public API and for the node ABI.
Do you agree that they should both go into `massa-proto-bindings-rs` in different folders? Or do you think we should separate them?

**Comment by ASAPSegfault** [comment]
Regarding the name I have no problem with massa-proto-bindings-xxx.

I think the Rust ABIs should go in the same repository, only in different folders.

**Comment by bilboquet** [comment]
# Overall organization
After some discussions today, we are unsure that `bindings` is a good name but we haven't found a better alternative so we go for `massa-proto-bindings`, at least for the moment.

We (@aoudiamoncef and I) aim to organize `massa-proto-bindings` so it provides the "generation scripts" and the generated bindings for the languages we support in dedicated directories in the suited format for the target language.
Hence, as an example for Rust we should have something like:
```
massa-proto-bindings/rust/
				 + Cargo.toml
				 + build.rs
				 + src/...

```
This way a client project for this bindings should be able to depends on it the `cargo` way. We're looking forward to see if it's ok for other languages (we're hoping for now). If it does not work we'll have to switch to the original proposition of one repository per language.

`massa-proto-bindings` should be automatically updated and tagged on event on `massa-proto` (push to `main` or tag).

# About the dependencies
One aim of `massa-proto` and `massa-proto-bindings` is to have unambiguous versioning for projects that depends on each other via the proto definitions.
Below are represented some of those dependencies (not complete, not fully up to date or may represent what we aim). 

[![](https://mermaid.ink/img/pako:eNqVkT9PwzAQxb-KZdZYDPwZjIRESWGBhW5tMhy2k1iJ7eh8UaiqfndCHFALEx6se_d-94a7A1dBGy551YVRNYDEXt4KX3g2ve3OQYxwOf-ixl6VTIh7tkr2ehbPCRIQRdTtGZDP4mkBcIgkovpDPc5ivRshOlZhcAxieWrlJxb-WEvqFIiDJ-vMWehqsd-t19bXMZkPS7fHQKFkCe2sbze07wy7YZEwtEZeVNVVlmoxWk2NvO4_MhW6gBKNvvs9ePufQZ5xZ9CB1dPaD1-dglNjnCm4nEptKhg6KnjhjxMKA4XN3isuCQeT8aHXQCa3UCM4Livo4tQ12lLA13TK-aIZ78FvQ_hmjp89n559?type=png)](https://mermaid.live/edit#pako:eNqVkT9PwzAQxb-KZdZYDPwZjIRESWGBhW5tMhy2k1iJ7eh8UaiqfndCHFALEx6se_d-94a7A1dBGy551YVRNYDEXt4KX3g2ve3OQYxwOf-ixl6VTIh7tkr2ehbPCRIQRdTtGZDP4mkBcIgkovpDPc5ivRshOlZhcAxieWrlJxb-WEvqFIiDJ-vMWehqsd-t19bXMZkPS7fHQKFkCe2sbze07wy7YZEwtEZeVNVVlmoxWk2NvO4_MhW6gBKNvvs9ePufQZ5xZ9CB1dPaD1-dglNjnCm4nEptKhg6KnjhjxMKA4XN3isuCQeT8aHXQCa3UCM4Livo4tQ12lLA13TK-aIZ78FvQ_hmjp89n559)

Main focus here is given to the ABI and `massa-sc-runtime`. At the time of writing, the yellow links does not really exist, at least they are weak (`wasm` files used for tests are "hard wired" in `massa-sc-runtime`).
If I'm correct, `.wasm` originated from https://github.com/massalabs/massa-unit-tests-src we should make `massa-sc-runtime` tests depend directly onto the tests sources. This way we should ensure that at a point in time `massa-sc-runtime` and `massa-*-sdk` that depends upon a version of the `.proto` files are in sync.

Same scheme should apply the API side but I don't know well this part.

@damip @ASAPSegfault @AurelienFT feel free to share your thoughts :)

**Reply by bilboquet** [reply]
Updated version of the dependencies (at compile time). 
Still not perfect but more precise.
Yellow link (that does not currently) still appears as a good way to ensure coherence of the system.

[![](https://mermaid.ink/img/pako:eNptk11rwyAUhv-KOLarhF1sMMhgUNq1--zFelFo0wsXT5rQqEFPaEvpf59RM9yHF-J53lfP8ZicaKE40IyWjdoXFdNI3j5ySewYrQUzhqWtVqg2no0DM0WqO4m1gCBMg6A7g71q-C4os6AwE8Gn-GwrBfz8A-sBv_zA26GY-Z9irhEMBnXl1Wvv2eq22JBwHEnTBzIK5bngKUZTFzzH6PKSjF38uN4zI0iplSDfZVv10amzX-5J5NaRe-LUaeipTxcHczcvw0Vi3c-m-9xq1lZkuV7aBDOQoBnWStoU3tGP19CfTtbYN2YooB9vv94LDky0DcSW9-jh_jX4FUge7Ynb4GrwgFwNt3XJh-s75HlTy90Cjw2QO2JQqx1kF2V5k_h1uq85Vtlte0gK1SidaeD3_UaaUAFasJrbb_jUk5xiBQJymtklh5J1DeY0l2drZR2qxVEWNEPdQUK7ljOESc1sNwXNStYYS4HXqPS7_y_c75HQlsmVUoPn_AXO__VP?type=png)](https://mermaid.live/edit#pako:eNptk11rwyAUhv-KOLarhF1sMMhgUNq1--zFelFo0wsXT5rQqEFPaEvpf59RM9yHF-J53lfP8ZicaKE40IyWjdoXFdNI3j5ySewYrQUzhqWtVqg2no0DM0WqO4m1gCBMg6A7g71q-C4os6AwE8Gn-GwrBfz8A-sBv_zA26GY-Z9irhEMBnXl1Wvv2eq22JBwHEnTBzIK5bngKUZTFzzH6PKSjF38uN4zI0iplSDfZVv10amzX-5J5NaRe-LUaeipTxcHczcvw0Vi3c-m-9xq1lZkuV7aBDOQoBnWStoU3tGP19CfTtbYN2YooB9vv94LDky0DcSW9-jh_jX4FUge7Ynb4GrwgFwNt3XJh-s75HlTy90Cjw2QO2JQqx1kF2V5k_h1uq85Vtlte0gK1SidaeD3_UaaUAFasJrbb_jUk5xiBQJymtklh5J1DeY0l2drZR2qxVEWNEPdQUK7ljOESc1sNwXNStYYS4HXqPS7_y_c75HQlsmVUoPn_AXO__VP)

**Reply by bilboquet** [reply]
@Leo-Besancon @Eitu33 for reference purpose
